test/unit/BCoWHelper.t.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.25;
+
+import {BCoWHelper} from 'contracts/BCoWHelper.sol';
+import {Test} from 'forge-std/Test.sol';
+
+import {IBCoWPool} from 'interfaces/IBCoWPool.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {ICOWAMMPoolHelper} from '@cow-amm/interfaces/ICOWAMMPoolHelper.sol';
+import {GPv2Interaction} from '@cowprotocol/libraries/GPv2Interaction.sol';
+import {GPv2Order} from '@cowprotocol/libraries/GPv2Order.sol';
+
+import {MockBCoWFactory} from 'test/manual-smock/MockBCoWFactory.sol';
+import {MockBCoWPool} from 'test/manual-smock/MockBCoWPool.sol';
+
+contract BCoWHelperTest is Test {
+  BCoWHelper helper;
+
+  MockBCoWFactory factory;
+  MockBCoWPool pool;
+  address invalidPool = makeAddr('invalidPool');
+  address[] tokens = new address[](2);
+  uint256[] priceVector = new uint256[](2);
+
+  uint256 constant VALID_WEIGHT = 1e18;
+
+  function setUp() external {
+    factory = new MockBCoWFactory(address(0), bytes32(0));
```

Comment:
i do think that deploying blank mocks and setting individual storages is preferrable, we need to note that:
- MockBCoWFactory doesn't deploy MockBCoWPools, lol
- we don't care what `solutionSettler` in BCoWFactory is set (zero is ok)
- we do care in BCoWPool (since is read on constructor)